src/components/application_manager/rpc_plugins/sdl_rpc_plugin/src/commands/mobile/reset_global_properties_request.cc

```diff
@@ -353,6 +152,56 @@ bool ResetGlobalPropertiesRequest::IsPendingResponseExist() {
          IsInterfaceAwaited(HmiInterfaces::HMI_INTERFACE_UI);
 }
 
+void ResetGlobalPropertiesRequest::ResetGlobalProperties() {
+  LOG4CXX_AUTO_TRACE(logger_);
+
+  uint32_t app_id =
+      (*message_)[strings::params][strings::connection_key].asUInt();
+  ApplicationSharedPtr app = application_manager_.application(app_id);
+
+  if (!app) {
+    LOG4CXX_ERROR(logger_, "No application associated with session key");
+    SendResponse(false, mobile_apis::Result::APPLICATION_NOT_REGISTERED);
+    return;
+  }
+
+  const auto& global_properties =
+      (*message_)[strings::msg_params][strings::properties];
+
+  auto reset_global_props_result =
+      application_manager_.ResetGlobalProperties(global_properties, app_id);
+
+  if (reset_global_props_result.HasUIPropertiesReset()) {
+    StartAwaitForInterface(HmiInterfaces::HMI_INTERFACE_UI);
```

Comment:
@iCollin no, it must be always before sending requests to HMI for understanding how many interfaces should be in the waitlist. Especially for cases when RPC can be split into more than one interface(UI, VR, TTS). This is necessary to avoid some kind of data races/crashes.
Imagine the following scenario:
Thread A: calls StartAwaitForInterface(UI)
Thread A: calls SendHMIRequest(UI) --> message sent out to HMI
Thread B: Immediately receives response from HMI for UI part and calls `on_event()` callback
Thread B: `on_event` verifies that there is no other interfaces in the waitlist and sends response to mobile and destroys the request object
Thread A: calls StartAwaitForInterface(VR)
Thread A: calls SendHMIRequest(VR) --> possible crash on attempt to access `message_` field of the destroyed request object / undefined behavior / weird further behavior